conversions/Forwards/src/main/java/com/agentdid127/resourcepack/forwards/impl/BlockStateConverter.java

```diff
@@ -90,95 +99,67 @@ public void convert(Pack pack) throws IOException {
-                    anyChanges = true;
+            if (value.has("model"))
+                updateModelObject(value);
+        } else if (entry.getValue() instanceof JsonArray) {
+            for (JsonElement jsonElement : ((JsonArray) entry.getValue())) {
+                if (jsonElement instanceof JsonObject) {
+                    JsonObject value = (JsonObject) jsonElement;
+                    if (value.has("model"))
+                        updateModelObject(value);
                 }
+            }
+        }
+    }
 
-                if (from < Util.getVersionProtocol(packConverter.getGson(), "1.19")
-                        && to >= Util.getVersionProtocol(packConverter.getGson(), "1.19")) {
-                    val = nameConverter.getBlockMapping19().remap(val);
-                    anyChanges = true;
-                }
+    private void updateModelObject(JsonObject value) {
+        NameConverter nameConverter = packConverter.getConverter(NameConverter.class);
 
-                if (from < Util.getVersionProtocol(packConverter.getGson(), "1.19.3")
-                        && to >= Util.getVersionProtocol(packConverter.getGson(), "1.19.3")) {
-                    prefix = "minecraft:" + prefix;
-                    anyChanges = true;
-                }
+        String[] split = value.get("model").getAsString().split("/");
+        String val = split[split.length - 1];
+        String prefix = value.get("model").getAsString().substring(0,
+                value.get("model").getAsString().length() - val.length());
 
-                if (anyChanges)
-                    value.addProperty("model", prefix + val);
-            }
-        } else if (entry.getValue() instanceof JsonArray) { // some states have arrays
-            for (JsonElement jsonElement : ((JsonArray) entry.getValue())) {
-                if (jsonElement instanceof JsonObject) {
-                    JsonObject value = (JsonObject) jsonElement;
-                    if (value.has("model")) {
-                        String[] split = value.get("model").getAsString().split("/");
-                        String val = split[split.length - 1];
-                        String prefix = value.get("model").getAsString().substring(0,
-                                value.get("model").getAsString().length() - val.length());
-
-                        if (from < Util.getVersionProtocol(packConverter.getGson(), "1.13")
-                                && to >= Util.getVersionProtocol(packConverter.getGson(), "1.13")) {
-                            val = nameConverter.getBlockMapping().remap(val);
-                            prefix = prefix.replaceAll("blocks", "block");
-                            anyChanges = true;
-                        }
+        if (from < Util.getVersionProtocol(packConverter.getGson(), "1.9")
+                && to >= Util.getVersionProtocol(packConverter.getGson(), "1.9")) {
+            prefix = "block/" + prefix;
```

Comment:
Do we know for certain that there are no other folders for block states?

Reply:
If others can exist, check for instances of the character / and don't do this if that occurs.